lib/internal/tls/wrap.js

```diff
@@ -1760,15 +1767,6 @@ exports.connect = function connect(...args) {
     tlssock.setSession(options.session);
 
   if (options.servername) {
-    if (!ipServernameWarned && net.isIP(options.servername)) {
-      process.emitWarning(
-        'Setting the TLS ServerName to an IP address is not permitted by ' +
-        'RFC 6066. This will be ignored in a future version.',
```

Comment:
Quite minor... but the warning says that "this will be ignored in a future version", while your changes are introducing an error for this... I don't know if it really matters I just figured I'd point out the discrepancy :slightly_smiling_face:

Reply:
I don't think it actually can be ignored if we're removing it. Ignoring a value that is passed as an IP opens up subtle difficult to diagnose bugs (I'm providing a servername, why isn't it working?).

Reply:
ok, sounds good to me, I do agree that it's better to have a more correct behavior than completely adhering to the existing warning message :slightly_smiling_face: